python/ray/tune/progress_reporter.py

```diff
@@ -621,3 +654,68 @@ def _get_trial_info(trial, parameters, metrics):
         unflattened_lookup(metric, result, default=None) for metric in metrics
     ]
     return trial_info
+
+
+class TrialProgressCallback(Callback):
+    def __init__(self, metric: Optional[str] = None):
+        self._last_print = collections.defaultdict(lambda: 0.)
+        self._metric = metric
+
+    def on_trial_result(self, iteration: int, trials: List["Trial"],
+                        trial: "Trial", result: Dict, **info):
+        self.log_result(trial, result, error=False)
+
+    def on_trial_error(self, iteration: int, trials: List["Trial"],
+                       trial: "Trial", **info):
+        self.log_result(trial, trial.last_result, error=True)
+
+    def log_result(self, trial: "Trial", result: Dict, error: bool = False):
+        done = result.get("done", False) is True
+        last_print = self._last_print[trial]
+        if has_verbosity(Verbosity.TRIAL_DETAILS) and \
+           (done or time.time() - last_print > DEBUG_PRINT_INTERVAL):
+            print("Result for {}:".format(self))
```

Comment:
Wouldn't this print a `Result for Progress Reporter callback` instead of referring to the trial?

Reply:
Right, thanks for spotting!